### DISCUSSION TITLE
Invitation links

### DISCUSSION THREAD

**Discussion by kevingzhang** [tearust/teaproject#35] [Ideas]
Invitation links is used for any existing user to invite their friends to join TEA Mining.

## The UI

Inviter can generate an invitation link in wallet UI. In this UI, he can optionally select a CML or CML seeds as a gift attached.  he can send this link to his friends "invitee" using any messaging system. 
Invitee need to create his own Polka extension and address. When get into the wallet UI, he can copy the invitation link in the UI. The invitation link will create 
- a 2000 Coffee credit in new invitees' account and
- a 2000 Coffee debt in the invitee' account.
- If the inviter optionally attached a CML in the invitation, the invitee will own this CML
There is neutral to the whole ecosystem since no new COFFEE token was mint.

The new invitee can use the 2000 COFFEE credit to start mining. 
But before the debt in COFFE is paid off, he cannot 
- transfer funds(TEA or COFFEE) to another account.

## Why would the inviter do that?

In the next epoch. the mining machine is no longer free. Miners need to pay COFFEE to buy mining machine to start mining even he owns a frosted seed. If the miner doesn't have enough COFFEE balance to buy new mining machine, but he has CML seeds. One possible solution is to give the CML seeds to his friends to join the mining. His friend, the invitee, doesn't need to own anything to start mining. His friend can also send the future revenue (after the invitee paid off the debt) to him.

If the inviter invested in some TApp, and would like to have more others to invest too, he can invite his friends. Especially those who have social influence.  

## Benefit to the TEA ecosystem
Using social viral to get more mining join TEA mining. No financial impact since we did not mint new NFT/FT.

**Comment by kevingzhang** [comment]
This feature requires the following development 
- COFFEE debt
- Invitation links UI
- Limit transfer when there is a COFFEE debt
- Redeem invitation links
- Transfer ownership of CML if it is attached to an invitation link

**Reply by geopor** [reply]
Although an invite feature could help grow participation (especially if there's an affiliate incentive for the person who's inviting others), I think we are a bit early and I'm not sure these will be used by our contestants.

> If the inviter invested in some TApp, and would like to have more others to invest too, he can invite his friends. Especially those who have social influence.

It sounds great in theory but consider that our contestant must have a friend who meets all of the following criteria:

a) has social influence particularly in crypto.
b) not afraid to install polkadot.js extension (will be new to 99% of people even in crypto).
c) understand basics of a bonding curve and how the token value of the video they're promoting grows as more people buy the token.
d) willing to join our contest.

I would hesitate to add this feature if it meant taking dev hours away from a different area.

**Reply by kevingzhang** [reply]
OK, let's postpone this feature to future epoch. 
We will need to find a way to incentivize the referrer or/and referee. if we find the incentivize, it is doable.
